podTemplate.yaml

```diff
@@ -0,0 +1,51 @@
+---
+apiVersion: "v1"
+kind: "Pod"
+spec:
+  containers:
+    - name: jnlp
+      image: jenkinsciinfra/jenkins-agent-ubuntu-22.04@sha256:35420f09777717099416fb80f334f5c5d34d09ac29743c12be21dbb0e1628f1f
+      imagePullPolicy: "IfNotPresent"
+      command:
+        - "/usr/local/bin/jenkins-agent"
+      env:
+        - name: "PATH"
+          value: "/opt/jdk-11/bin:/home/jenkins/.asdf/shims:/home/jenkins/.asdf/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+        - name: "ARTIFACT_CACHING_PROXY_PROVIDER"
+          value: "aws"
```

Comment:
Does this get overridden?

Reply:
What do you mean?

Reply:
IIUC https://github.com/jenkinsci/bom/blob/57116da6cd96ed5f0cc4abd45ea17be64bf98570/Jenkinsfile#L18 should set the same variable?

Reply:
We're setting this `ARTIFACT_CACHING_PROXY_PROVIDER` variable in every agent (except the Azure VM agents as it's not possible to set an env var on them AFAIK) in our infrastructure as code definitions.
This variable is then used in the `infra.withArtifactCachingProxy` function to determine which acp provider to use.
This variable is not set in the pipeline library.
Putting it in this podTemplate allows the correct selection of the AWS acp provider, without it it would default to the Azure one. (cf https://github.com/jenkins-infra/pipeline-library/blob/9e0f0a2f09a7c91b26824e706a8963598b685742/vars/infra.groovy#L101)